flocker/control/_protocol.py

```diff
@@ -350,7 +351,12 @@ def cancel(self):
         """
         Cancel the delayed call to this ``Timeout``'s ``action``.
         """
-        self._delay_call.cancel()
+        try:
+            self._delay_call.cancel()
+        except AlreadyCalled:
+            # This Timeout may have triggered protocol.abortConnection which
+            # will attempt to cancel...this timeout...
+            pass
```

Comment:
by `pass`ing if already called, how does the dataset agent eventually reconnect? Simply ignoring the call the agent will eventually not fail out and reconnect?